grunttasks/scan-email-translations.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  grunt.registerTask('scan-email-translations', 'Scan the generated emails for invalid style tags', [
+    'generate-valid-styles',
+    'verify-styles'
+    ]
+  );
+
+  grunt.registerMultiTask('verify-styles', 'Verify whether the translated styles are valid', function (){
+    var fileArray = this.files[0].src;
+    var contents = grunt.file.read(fileArray.shift());
+    var styleArray = JSON.parse(contents);
+
+    var hasErrors = 0;
+    var styleRegex = /style='(.*)'/ig;
+
+    fileArray.forEach(function (src) {
+      var html = grunt.file.read(src);
+      var styleList = html.match(styleRegex);
+      // Check if there were any styles found, .match() might return null
+      if (styleList !== null) {
+        styleList.forEach(function (style) {
+          if (styleArray.indexOf(style) === -1) {
+            grunt.log.error('Found a style which should not be there', style, src);
+            hasErrors = hasErrors + 1;
+          }
+        });
+      }
+    });
+    if (hasErrors !== 0) {
+      grunt.fail.warn('Found ' + hasErrors + ' ' + grunt.util.pluralize(hasErrors, 'error/errors') + ' in ' + fileArray.length + ' files');
+    } else {
+      grunt.log.writeln('Checked ' + fileArray.length + ' files for valid styles and found nothing wrong.');
+    }
+  });
+
+  grunt.config('verify-styles', {
+    dist: {
+      files: [
+        {
+          expand: false,
+          src: [
+            '.tmp/validStyles.json',
+            'locale/*/*/server.po'
+          ]
+        }
+      ]
+    }
+  });
+
+
+  grunt.registerMultiTask('generate-valid-styles', 'Generate the list of valid styles for translation', function () {
+    var styleRegex = /style='(.*)'/ig;
```

Comment:
See https://github.com/mozilla/fxa-content-server-l10n/pull/62/files#r35603540

Also, since this RegExp is duplicated, we may want to see if we can DRY it out.